app/src/main/java/com/yalta/utils/UniversalRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package com.yalta.utils
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+
+class UniversalRecyclerViewAdapter : RecyclerView.Adapter<BindingViewHolder>() {
+    private val items = mutableListOf<UniversalRecyclerItem>()
+
+    override fun getItemCount(): Int {
+        return items.size
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return getItem(position).layoutId
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): BindingViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding: ViewDataBinding = DataBindingUtil.inflate(inflater, viewType, parent, false)
+        return BindingViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(
+        holder: BindingViewHolder,
+        position: Int
+    ) {
+        getItem(position).bind(holder.binding)
+        holder.binding.executePendingBindings()
+    }
+
+    fun updateData(newItems: List<UniversalRecyclerItem>) {
+        this.items.clear()
+        this.items.addAll(newItems)
```

Comment:
HMMMMMM, let's leave it for now, but I bet this is slow and visually laggy (and also maybe drops scrollview position)

Reply:
I have a strong opinion that it works like it should...
I mean
It's going to be updated by pulling down (I'll do it a bit later), so it's fine for everything to disappear for a moment